print_nanny_webapp/subscriptions/views.py

```diff
@@ -0,0 +1,172 @@
+from django.http.response import JsonResponse
+import stripe
+import json, logging
+from django.urls import reverse
+from django.http import HttpRequest, request
+from django.contrib.auth import get_user_model
+from django.shortcuts import redirect
+from djstripe import webhooks
+
+import djstripe.models
+import djstripe.settings
+from anymail.message import AnymailMessage
+from django.template.loader import render_to_string
+
+from .models import start_trial
+from print_nanny_webapp.dashboard.views import DashboardView
+from print_nanny_webapp.remote_control.models import OctoPrintDevice
+
+logger = logging.getLogger(__name__)
+User = get_user_model()
+
+
+class SubscriptionsListView(DashboardView):
+    template_name = "subscriptions/list.html"
+    # template_name = "components/base-ui/cards.html"
+
+    def get_context_data(self, *args, **kwargs):
+        ctx = super().get_context_data(*args, **kwargs)
+
+        try:
+            customer = djstripe.models.Customer.objects.get(subscriber=self.request.user)
+        except djstripe.models.Customer.DoesNotExist:
+            start_trial(self.request.user, instance=self.request.user, created=True)
+            customer = djstripe.models.Customer.objects.get(subscriber=self.request.user)
+
+        # Populate template with current subscriptions and stripe public key
+        ctx["STRIPE_PUBLIC_KEY"] = djstripe.settings.STRIPE_PUBLIC_KEY
+        ctx["SUBSCRIPTIONS"] = customer.subscriptions.all().order_by('-created')
+        ctx["PRODUCTS"] = djstripe.models.Product.objects.filter(active=True)
+        for p in ctx["PRODUCTS"]:
+            p.prices_list = p.prices.filter(active=True)
+
+        try:
+            ctx["DEVICES"] = OctoPrintDevice.objects.filter(user=self.request.user)
```

Comment:
`OctoPrintDevice.objects.filter` will return an empty list `[]` here. If you'd like to raise an `OctoPrintDevice.DoesNotExist` exception, use `OctoPrintDevice.objects.get`

```
$ docker-compose -f local.yml run --rm django python manage.py shell
Type 'copyright', 'credits' or 'license' for more information
IPython 7.21.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from django.apps import apps

In [2]: OctoPrintDevice = apps.get_model("remote_control", "OctoPrintDevice")
In [4]: OctoPrintDevice.objects.filter(user=100)
Out[4]: <QuerySet []>

In [5]: OctoPrintDevice.objects.get(user=100)
---------------------------------------------------------------------------
DoesNotExist                              Traceback (most recent call last)
<ipython-input-5-f4970364ee52> in <module>
----> 1 OctoPrintDevice.objects.get(user=100)

/usr/local/lib/python3.8/site-packages/django/db/models/manager.py in manager_method(self, *args, **kwargs)
     83         def create_method(name, method):
     84             def manager_method(self, *args, **kwargs):
---> 85                 return getattr(self.get_queryset(), name)(*args, **kwargs)
     86             manager_method.__name__ = method.__name__
     87             manager_method.__doc__ = method.__doc__

/usr/local/lib/python3.8/site-packages/django/db/models/query.py in get(self, *args, **kwargs)
    427             return clone._result_cache[0]
    428         if not num:
--> 429             raise self.model.DoesNotExist(
    430                 "%s matching query does not exist." %
    431                 self.model._meta.object_name

DoesNotExist: OctoPrintDevice matching query does not exist.
```